src/user/client.ts

```diff
@@ -65,28 +71,80 @@ const client = {
       });
   },
   googleLogin() {
-    const provider = new firebase.auth.GoogleAuthProvider();
-    return firebase
-      .auth()
-      .signInWithPopup(provider)
-      .then(result => {
-        if (result.credential) {
-          const credential = result.credential as firebase.auth.OAuthCredential;
-          const token = credential.accessToken;
-          console.log("OK - OAuth Token: " + token);
-        }
-        if (result.user != null) {
-          db.ref("users/" + result.user.uid).once("value", async snapshot => {
-            if (!snapshot.val() && result.user) {
-              const newUser = createUserObject(result.user);
-              saveUserToDb(newUser);
+    return new Promise((resolve, reject) => {
+      gapi.load("auth2", async () => {
+        gapi.auth2
+          .init({
+            // eslint-disable-next-line @typescript-eslint/camelcase
+            client_id: googleCalendarClient.clientId,
+            scope: googleCalendarClient.scope
+          })
```

Comment:
I think it would be good to abstract this logic out into a function or module so we only have to do it once per load.